nbtc/relayer_test.go

```diff
@@ -55,49 +64,35 @@ func Test_Start(t *testing.T) {
 	relayer.db.Close()
 }
 
-func Test_processSignedTxs(t *testing.T) {
+func Test_NewRelayer_DatabaseError(t *testing.T) {
 	db := initTestDB(t)
-	daltest.PopulateSignRequests(t, db)
-	daltest.PopulateBitcoinTxs(t, db)
-	relayer, err := NewRelayer(btcClientConfig, relayerConfig, db)
-	assert.NilError(t, err)
-	relayer.btcClient = &bitcoin.MockClient{}
-
-	err = relayer.processSignedTxs()
-	assert.NilError(t, err)
+	mockIkaClient := ika.NewMockClient()
+	btcProcessor, _ := ika2btc.NewProcessor(btcClientConfig, 6, db)
+	btcProcessor.BtcClient = &bitcoin.MockClient{}
+	nativeProcessor := native2ika.NewProcessor(mockIkaClient, db)
 
-	updatedTx, err := db.GetBitcoinTx(2, daltest.GetHashBytes(t, "0"))
-	assert.NilError(t, err)
-	assert.Equal(t, updatedTx.Status, dal.Broadcasted)
+	relayer, err := NewRelayer(relayerConfig, nil, nativeProcessor, btcProcessor)
+	assert.ErrorContains(t, err, "database cannot be nil")
+	assert.Assert(t, relayer == nil)
 }
 
-func Test_checkConfirmations(t *testing.T) {
+func Test_NewRelayer_NativeProcessorError(t *testing.T) {
```

Comment:
let's merge this test with the one above

Reply:
Done